data_pipeline/civicrm_email_report/civicrm_email_report.py

```diff
@@ -0,0 +1,65 @@
+import json
+import datetime
+import logging
+from typing import Iterable
+import requests
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_current_timestamp():
+    return datetime.datetime.utcnow().isoformat()
+
+
+def get_mailing_id_dict(mail_id: int) -> str(dict):
+    return json.dumps({'mailing_id': mail_id})
+
+
+def get_connection_parameters(mail_id: int, api_key, site_key):
+    return {
+        "entity": "Mailing",
+        "action": "stats",
+        "json": get_mailing_id_dict(mail_id),
+        "api_key": api_key,
+        "key": site_key
+    }
+
+
+def get_email_report(
+    url: str,
+    mail_id: int,
+    api_key: str,
+    site_key: str
+) -> dict:
+    response = requests.post(url=url, data=get_connection_parameters(mail_id, api_key, site_key))
+    response.raise_for_status()
+    dict_response = json.loads(response.text)
+    mail_id_str = str(mail_id)
+    return {
+        "mail_id": mail_id,
+        "delivered": int(dict_response["values"][mail_id_str]["Delivered"]),
+        "delivered_rate": float(
+            dict_response["values"][mail_id_str]["delivered_rate"].replace("%", "")
+        ),
+        "bounces": int(dict_response["values"][mail_id_str]["Bounces"]),
+        "unsubscribers": int(dict_response["values"][mail_id_str]["Unsubscribers"]),
+        "unique_clicks": int(dict_response["values"][mail_id_str]["Unique Clicks"]),
+        "opened": int(dict_response["values"][mail_id_str]["Opened"]),
+        "opened_rate": float(dict_response["values"][mail_id_str]["opened_rate"].replace("%", "")),
+        "clickthrough_rate": float(
+            dict_response["values"][mail_id_str]["clickthrough_rate"].replace("%", "")
+        ),
+        "imported_timestamp": get_current_timestamp()
+    }
+
+
+def iter_email_report(
+    url: str,
+    mail_id_list: list,
```

Comment:
I would maybe add a more specific type. e.g. `Sequence[str]`.

Currently it seems that we are passing a list of lists / tuples, whereas the mail id as at index `1`.
Could we just pass in the list of mail ids?

Reply:
We will need to create different function to query it from BQ. But I prefer to use existing one and we are using this function in different pipelines too. 

What is `Sequence[str]` for? mail_id_list type?

Reply:
Currently it would be `Sequence[Sequence[str]]` or `Sequence[Tuple[str, str]]` or could also be `List[List[str]]`. I don't know what the types of the items are. `Sequence` is often preferable over `List` as an input type as it is less restrictive. We could also accept `Iterable` if we didn't use `len`.

Reply:
Now it is just a `list`.